docs/attributes-registry/log.md

```diff
@@ -40,12 +40,71 @@ Attributes for a file to which log was emitted.
 
 This document defines the generic attributes that may be used in any Log Record.
 
-| Attribute             | Type   | Description                                 | Examples                                                                                                                         | Stability                                                        |
-| --------------------- | ------ | ------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------- |
-| `log.record.original` | string | The complete orignal Log Record. [1]        | `77 <86>1 2015-08-06T21:58:59.694Z 192.168.2.133 inactive - - - Something happened`; `[INFO] 8/3/24 12:34:56 Something happened` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
-| `log.record.uid`      | string | A unique identifier for the Log Record. [2] | `01ARZ3NDEKTSV4RRFFQ69G5FAV`                                                                                                     | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
-
-**[1]:** This value MAY be added when processing a Log Record which was originally transmitted as a string or equivalent data type AND the Body field of the Log Record does not contain the same value. (e.g. a syslog or a log record read from a file.)
-
-**[2]:** If an id is provided, other log records with the same id will be considered duplicates and can be removed safely. This means, that two distinguishable log records MUST have different values.
-The id MAY be an [Universally Unique Lexicographically Sortable Identifier (ULID)](https://github.com/ulid/spec), but other identifiers (e.g. UUID) may be used as needed.
+| Attribute             | Type   | Description                                  | Examples                                                                                                                         | Stability                                                        |
+| --------------------- | ------ | -------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------- |
+| `log.record.id`       | int    | A durable identifier for the Log Record. [1] | `1`; `10018`                                                                                                                     | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```

Comment:
Are there any uniqueness requirements/guarantees for the id? Do we expect it to be globally unique? Unique within some limited scope? Are there any mechanisms for preventing id conflicts?

We also have `code.*` attributes that can point to a particular place in the code that emits the log record that can serve as sort of durable identifier (e.g. filename+function+line). What's the interplay (if any) between these and the newly proposed attribute?

Reply:
filename/function/line etc. is not really durable, as it can change just with code refactoring!

Here is an example of EventId/Name: https://learn.microsoft.com/en-us/dotnet/core/extensions/logging?tabs=command-line#log-event-id

Reply:
OK, makes sense regarding code references.

I think we still need clarity around uniqueness, conflicts, etc.

Reply:
Resolving this conversation and we can continue in https://github.com/open-telemetry/semantic-conventions/pull/1339#discussion_r1729415678